### PR TITLE
common constant function

### LINKED ISSUE #618: Refactor MOI._constant and MOIU.getconstant
Create good names consistent with the [JuMP style guide](http://www.juliaopt.org/JuMP.jl/v0.19.0/style/)

### PR DESCRIPTION
**Pull request by matbesancon**
Merge `MOI._constant` and `MOIU.getconstant`. Fixes #618

### CODE REVIEW

**Comment by blegat** [comment]
We might also add a way to get zero for the constant of `SingleVariable` and `VectorOfVariables` to solve the use case in:
https://github.com/JuliaOpt/MathOptInterface.jl/blob/c1d8f6a6949a39091c89d8c4053d51a7f6cc8d03/src/Bridges/scalarizebridge.jl#L7-L10
For that we need to know what is the coefficient type. We could allow a second argument which is the coefficient type but is optional for every set/function except `SingleVariable`/`VectorOfVariables`.

For the name, `getconstant` does not satisfy JuMP style guide, it should be `constant`, `get_constant`, or `set_constant`/function_constant`.
The second choice is inconsistent with the naming in the rest of JuMP/MOI but the first and third choices works (the first one is better IMO since the fact that it is a `set` or `function` is already in the signature so adding it in the name is redundant).

File: src/Utilities/sets.jl [hunk 1]
```diff
@@ -3,9 +3,9 @@
 
 Returns the constant of the set.
 """
-getconstant(s::MOI.EqualTo) = s.value
-getconstant(s::MOI.GreaterThan) = s.lower
-getconstant(s::MOI.LessThan) = s.upper
+MOI.getconstant(s::MOI.EqualTo) = s.value
```
Comment (blegat) [inline src/Utilities/sets.jl]: We can move the implementation to `MOI` if it's an MOI function.

File: src/functions.jl [hunk 2]
```diff
@@ -352,14 +352,42 @@ function _dicts(f::Union{ScalarQuadraticFunction, VectorQuadraticFunction})
...
+    isapprox(constant(f), constant(g); kwargs...) && all(dict_isapprox.(_dicts(f), _dicts(g); kwargs...))
 end
 
+constant(f::Union{ScalarAffineFunction, ScalarQuadraticFunction}, T::DataType) = constant(f)
+constant(f::Union{VectorAffineFunction, VectorQuadraticFunction}, T::DataType) = constant(f)
+
+"""
+    constant(f::SingleVariable, T::DataType)
+
+The constant term of a `SingleVariable` function is zero.
```
Comment (blegat) [inline src/functions.jl]: `T` should appear in the docstring

**Comment by blegat** [comment]
`constant` could appear somewhere in the doc, e.g. after http://www.juliaopt.org/MathOptInterface.jl/dev/apireference/#MathOptInterface.output_dimension

**Comment by matbesancon** [comment]
I put it after `output_dimension`

File: docs/src/apireference.md [hunk 3]
```diff
@@ -216,6 +216,7 @@ Functions for getting and setting properties of sets.
 
 ```@docs
 output_dimension
+constant
```
Comment (blegat) [inline docs/src/apireference.md]: There is a way with Documenter to only make methods with functions appear here and methods with sets appear after
http://www.juliaopt.org/MathOptInterface.jl/dev/apireference/#MathOptInterface.dimension

Comment (matbesancon) [reply docs/src/apireference.md]: I didn't find anything for this in Documenter's doc, do you have the syntax for that? The only solution I see is a really ugly hack using `Filters`

Comment (matbesancon) [reply docs/src/apireference.md]: And didn't find anything like it in the rest of the docs

Comment (blegat) [reply docs/src/apireference.md]: Normally, you can do something like `constant(::MathOptInterface.AbstractSet)`

Comment (blegat) [reply docs/src/apireference.md]: See https://juliadocs.github.io/Documenter.jl/stable/man/guide/#Filtering-included-docstrings-1 and https://github.com/JuliaDocs/Documenter.jl/issues/569

Comment (matbesancon) [reply docs/src/apireference.md]: ok I managed to fix it. It's a very tricky process I wouldn't advise using it whenever it's avoidable

**Comment by matbesancon** [comment]
ok this looks good

File: src/Utilities/sets.jl [hunk 4]
```diff
@@ -1,11 +1,3 @@
-"""
-    getconstant(s::Union{MOI.EqualTo, MOI.GreaterThan, MOI.LessThan})
-
-Returns the constant of the set.
-"""
-getconstant(s::MOI.EqualTo) = s.value
-getconstant(s::MOI.GreaterThan) = s.lower
-getconstant(s::MOI.LessThan) = s.upper
```
Comment (blegat) [inline src/Utilities/sets.jl]: Remove this line

**Comment by matbesancon** [comment]
@blegat seems good here

**Pull request merged**